src/org/zaproxy/zap/extension/httpsinfo/MenuEntry.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.httpsinfo;
+
+import java.awt.Component;
+
+import javax.swing.JTree;
+
+import org.apache.commons.httpclient.URIException;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.model.SiteNode;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.view.messagecontainer.http.HttpMessageContainer;
+import org.zaproxy.zap.view.popup.PopupMenuItemHttpMessageContainer;
+import org.zaproxy.zap.view.table.HistoryReferencesTable;
+
+/**
+ * This class defines the extension's popup menu item.
+ */
+public class MenuEntry extends PopupMenuItemHttpMessageContainer {
+
+	private static final long serialVersionUID = 1L;
+	private final ExtensionHttpsInfo extension;
+
+	private static final Logger LOGGER = Logger.getLogger(ExtensionHttpsInfo.class);
+
+	public MenuEntry(String label, ExtensionHttpsInfo extension) {
+		super(label);
+		this.extension = extension;
+	}
+
+	@Override
+	public boolean isEnableForInvoker(Invoker invoker, HttpMessageContainer httpMessageContainer) {
+		Component component = httpMessageContainer.getComponent();
```

Comment:
In this method you can use the `invoker` to check the component (instead of inspecting its type and contents).
For example, to show the pop up menu item just in Sites and History tabs you could do:
```Java
return invoker == Invoker.SITES_PANEL || invoker == Invoker.HISTORY_PANEL;
```

And then override the method isButtonEnabledForSelectedHttpMessage:
```Java
@Override
protected boolean isButtonEnabledForSelectedHttpMessage(HttpMessage message) {
    return message.getRequestHeader().isSecure();
}
```